ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Authorizer.scala

```diff
@@ -270,32 +269,31 @@ final class Authorizer(
               )
               .asGrpcError
           )
-        case claims: ClaimSet.Claims => Success(claims)
-      })
+        case claims: ClaimSet.Claims => Right(claims)
+      }
 
   private def authorizeWithReq[Req, Res](call: (Req, ServerCallStreamObserver[Res]) => Unit)(
       authorized: (ClaimSet.Claims, Req) => Either[StatusRuntimeException, Req]
   ): (Req, StreamObserver[Res]) => Unit = (request, observer) => {
     val serverCallStreamObserver = assertServerCall(observer)
-    authenticatedClaimsFromContext()
-      .fold(
-        ex => {
-          observer.onError(ex)
-        },
-        claims =>
-          authorized(claims, request) match {
-            case Right(modifiedRequest) =>
-              call(
-                modifiedRequest,
-                if (claims.expiration.isDefined || claims.resolvedFromUser)
-                  ongoingAuthorization(serverCallStreamObserver, claims)
-                else
-                  serverCallStreamObserver,
-              )
-            case Left(ex) =>
-              observer.onError(ex)
-          },
-      )
+    authenticatedClaimsFromContext() match {
+      case Left(e: StatusRuntimeException) => observer.onError(e)
```

Comment:
Because we're using `Either[StatusRuntimeException, A]` rather than `Try[A]` we can be precisely what exception we're expecting. If the exception type would be widen for whatever reason then the compiler will warn about non-exhaustive pattern matching